src/Symfony/Component/Validator/Constraints/UniqueCollectionValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Yevgeniy Zholkevskiy <zhenya.zholkevskiy@gmail.com>
+ */
+class UniqueCollectionValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof UniqueCollection) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\UniqueCollection');
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!is_array($value) && !($value instanceof \Traversable && $value instanceof \ArrayAccess)) {
+            throw new UnexpectedTypeException($value, 'array or Traversable and ArrayAccess');
+        }
+
+        $collectionElements = [];
+        foreach ($value as $element) {
+            if (in_array($element, $collectionElements)) {
```

Comment:
third argument to true is important here to avoid obscure bugs https://3v4l.org/9QGNo